numpy/_core/__init__.py

```diff
@@ -22,29 +22,60 @@
     from . import multiarray
 except ImportError as exc:
     import sys
-    msg = """
+
+    # Basically always, the problem should be that the C module is wrong/missing...
+    if (
+        isinstance(exc, ModuleNotFoundError)
+        and exc.name == "numpy._core._multiarray_umath"
+    ):
+        import sys
+        candidates = []
+        for path in __path__:
+            candidates.extend(
+                f for f in os.listdir(path) if f.startswith("_multiarray_umath"))
+        if len(candidates) == 0:
+            bad_c_module_info = (
+                "We found no compiled module, was NumPy build successfully?\n")
```

Comment:
tiny nit as I really shouldn't even be here 😄 :
```suggestion
                "We found no compiled module, did NumPy build successfully?\n")
```

Reply:
Sounds good, I guess I meant "was NumPu built", grammar is hard :).